core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -169,6 +170,29 @@ sealed abstract class Chunk[+A] extends ChunkLike[A] with Serializable { self =>
   def toPackedLong(endianness: Chunk.BitChunk.Endianness)(implicit ev: A <:< Boolean): Chunk[Long] =
     Chunk.ChunkPackedBoolean[Long](self.asInstanceOf[Chunk[Boolean]], 64, endianness)
 
+  /**
+   * Crates a new String based on this chunk of bytes.
+   */
+  final def asString(implicit ev: A <:< Byte): String =
+    asString(Charset.defaultCharset)
+
+  /**
+   * Crates a new String based on this chunk of bytes and using the given
+   * charset.
+   */
+  final def asString(charset: Charset)(implicit ev: A <:< Byte): String = {
+    implicit val cls: ClassTag[A] = classTag[Byte].asInstanceOf[ClassTag[A]]
+    new String(self.toArray.asInstanceOf[Array[Byte]], charset)
+  }
+
+  /**
+   * Crates a new String based on this chunk of chars.
+   */
+  final def asStringChar(implicit ev: A <:< Char): String = {
```

Comment:
It may be worth having:

```scala
object Chunk {
  ...
  sealed trait IsText[A] {
    def convert(chunk: Chunk[A]): String
  }
  object IsText {
    implicit val byteIsText: IsText[Byte] = ???
    implicit val charIsText: IsText[Char] = ???
    implicit val strIsText: IsText[String] = ???
  }
  ...
}
```

Then you can have one `asString` function:

```scala
def asString(implicit ev: isText: IsText[A]): String = ev.convert(self)
```

Aside from the overload for `Charset`, which is necessary.